tests/libnet/validation.go

```diff
@@ -52,3 +53,30 @@ func ValidateVMIandPodIPMatch(vmi *v1.VirtualMachineInstance, vmiPod *k8sv1.Pod)
 
 	return nil
 }
+
+// AssertAllPodIPsReportedOnVMI checks that each Pod IP is present in the VMI reported IPs
+func AssertAllPodIPsReportedOnVMI(vmi *v1.VirtualMachineInstance, vmiPod *k8sv1.Pod) error {
+	if vmi.Status.Interfaces[0].IP != vmiPod.Status.PodIP {
+		return fmt.Errorf("VMI Status.Interfaces[0].IP %s does not equal pod's Status.PodIP %s",
+			vmi.Status.Interfaces[0].IP, vmiPod.Status.PodIP)
+	}
+
+	// Ensure each Pod IP is present in the VMI reported IP set.
+	vmiIPSet := IPSliceToSet(vmi.Status.Interfaces[0].IPs)
+	for _, podIP := range vmiPod.Status.PodIPs {
+		_, exists := vmiIPSet[podIP.IP]
+		Expect(exists).To(BeTrue(),
+			fmt.Sprintf("Expected Pod IP %s to be present in VMI IPs %v",
+				podIP.IP, vmi.Status.Interfaces[0].IPs))
+	}
+	return nil
```

Comment:
Is it preferred to not use ginkgo expression inside helpers (although unfortunately there are many examples of it).
```suggestion
	for _, podIP := range vmiPod.Status.PodIPs {
		if _, exists := vmiIPSet[podIP.IP]; !exists {
		  return fmt.Sprintf("Expected Pod IP %q to be present in VMI IPs %v", podIP.IP,  vmi.Status.Interfaces[0].IPs))
		}		
	}
```